<?xml version="1.0"?>

<launch>
  <!--include file="$(find gazebo_ros)/launch/empty_world.launch"/-->

  <!--include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" default="$(find service_utd)/worlds/servicebatch.world"/>
    <arg name="headless" value="true" />
    <arg name="gui" value="false" />
  </include-->
  <param name="/use_sim_time" value="true" />

  <group ns="target">


        <!-- <rosparam param="ugs/x">[26.8,27.96,34.04,20.24,25.71,11.29]</rosparam> -->
				<!-- <rosparam param="ugs/y">[24,20.08,26.11,22.30,28.31,7.51]</rosparam> -->
        <!-- <rosparam param="ugs/x">[7.95,25.71,32.95,14.07,34.04,10.04,21.33,29.07,11.29,22.91]</rosparam> -->
				<!-- <rosparam param="ugs/y">[23.50,28.31,30.22,20.63,26.11,15.18,18.25,16.15,7.51,10.75]</rosparam> -->
        <rosparam param="ugs/x">[32.25,22.40,38.04,8.51,27.41,20.79,13.65,27.53,26.50,12.18]</rosparam>
        <rosparam param="ugs/y">[32.23,27.40,27.21,21.38,22.11,20.28,16.17,15.73,11.76,4.67]</rosparam>
				<param name="detection_radius" value="1" />
				<param name="comm_radius" value="2" />
				<param name="alpha" value="0.95" />
				<param name="beta" value="0.01" />
				<rosparam param="UAVconnectivity">[1,2,3,4]</rosparam>
        <param name="mapname" value="$(find service_utd)/matlabmap.yml" />
        <node name="fakerover" pkg="service_utd" type="fakerover" output="screen" />
        <param name="odometry_topic" value="p3dx/base_pose_ground_truth" />
        <node name="rover_controller" pkg="service_utd" type="rover_controller" output="screen">
        </node>
				<!-- <node name="groundsensors" pkg="service_utd" type="groundsensors" output="screen" /> -->


      </group>

      <group ns="uav1">
        <param name="worldframe" value="/world" />
        <param name="baselink" value="/uav1/base_link" />
        <param name="targetdetect/camera_info_url" value="$(find service_utd)/calibration.yml" />
        <param name="tf_prefix" value="uav1" />

        <rosparam param="droneinit">[-1.0,24.362551,1,0]</rosparam>
        <node name="fakedrone" pkg="service_utd" type="fakedrone" output="screen" />
        <rosparam param="drone_setpoint/gains">[0.4,0.4,0.8,0.8,0.8,1.6,0.8]</rosparam>
        <node name="drone_setpoint" pkg="service_utd" type="drone_setpoint" respawn="true" />

        <param name="target_topic" value="/target/p3dx/base_pose_ground_truth" />
        <param name="targetdetect/targetsize" value="3.0" />
        <node name="fakedetector" pkg="service_utd" type="fakedetector" respawn="true" />


        <param name="waypoints/x" value="-2 2 2 -2 -2 2" />
        <param name="waypoints/y" value="1.5 1.5 0 0 -1.5 -1.5" />
        <param name="waypoints/z" value="2 2 2 2 2 2" />
        <param name="waypointNav" value="false" type="bool" />
        <param name="alpha" value="0.95" />
        <param name="beta" value="0.01" />
        <param name="detection_radius" value="$(arg detection_radius)" />
        <param name="collision_radius" value="$(arg collision_radius)" />
        <param name="num_particles" value="4000" />
        <param name="mapname" value="$(find service_utd)/matlabmap.yml" />
        <rosparam param="UAVdetectability">[2,3,4,5,6]</rosparam>
        <rosparam param="UAVconnectivity">[2,3,4,5,6]</rosparam>
        <param name="base_altitude" value="4" />
        <param name="batchsimulation" value="true" />
        <param name="memsize" value="$(arg memsize)" />
        <param name="plan_length" value="$(arg plan_length)" />
        <param name="displaymap" value="false" />
        <node name="pathplanner" pkg="service_utd" type="pathplanner" required="true" output="screen"/>

        <param name="controller/twist/linear/xy/k_d" value="0.0" />
        <param name="controller/twist/linear/xy/k_i" value="0.5" />
        <param name="controller/twist/linear/xy/k_p" value="2.0" />

      </group>

      <group ns="uav2">
        <param name="targetdetect/camera_info_url" value="$(find service_utd)/calibration.yml" />
        <param name="tf_prefix" value="uav2" />

        <rosparam param="droneinit">[42.280787,38.019486,1,0]</rosparam>
        <node name="fakedrone" pkg="service_utd" type="fakedrone" output="screen" />
        <rosparam param="drone_setpoint/gains">[0.4,0.4,0.8,0.8,0.8,1.6,0.8]</rosparam>
        <node name="drone_setpoint" pkg="service_utd" type="drone_setpoint" respawn="true" />

        <param name="worldframe" value="/world" />
        <param name="baselink" value="/uav2/base_link" />

        <param name="target_topic" value="/target/p3dx/base_pose_ground_truth" />
        <param name="targetdetect/targetsize" value="3.0" />
        <node name="fakedetector" pkg="service_utd" type="fakedetector" respawn="true" />

        <param name="waypoints/x" value="-2 2 2 -2 -2 2" />
        <param name="waypoints/y" value="1.5 1.5 0 0 -1.5 -1.5" />
        <param name="waypoints/z" value="2 2 2 2 2 2" />
        <param name="waypointNav" value="false" type="bool" />
        <param name="alpha" value="0.95" />
        <param name="beta" value="0.01" />
        <param name="detection_radius" value="$(arg detection_radius)" />
        <param name="collision_radius" value="$(arg collision_radius)" />
        <param name="num_particles" value="4000" />
        <param name="mapname" value="$(find service_utd)/matlabmap.yml" />
        <rosparam param="UAVdetectability">[1,3,4,5,6]</rosparam>
        <rosparam param="UAVconnectivity">[1,3,4,5,6]</rosparam>
        <param name="base_altitude" value="4" />
        <param name="batchsimulation" value="true" />
        <param name="memsize" value="$(arg memsize)" />
        <param name="plan_length" value="$(arg plan_length)" />
        <param name="displaymap" value="false" />
        <node name="pathplanner" pkg="service_utd" type="pathplanner" required="true" output="screen"/>

        <param name="controller/twist/linear/xy/k_d" value="0.0" />
        <param name="controller/twist/linear/xy/k_i" value="0.5" />
        <param name="controller/twist/linear/xy/k_p" value="2.0" />


      </group>

    </launch>
